File: app/src/test/java/net/squanchy/speaker/domain/view/SpeakerFixtures.java
package net.squanchy.speaker.domain.view;

import net.squanchy.support.lang.Optional;

public final class SpeakerFixtures {

    private String id = "bananana";
    private long numericId = 5466L;
    private String name = "Banana Joe";
    private String bio = "Joe ¡oh Banana Joe!\nTu tienes, ¡oh Banana nana Joe!\nCorazón gigante y alma soñante\n¡Oh Banana nana Joe!\nJoe ¡oh Banana Joe!\nTu eres, ¡oh Banana na na Joe!\nUn gran marinero con puños de acero";
    private Optional<String> photoUrl = Optional.of("https://i.ytimg.com/vi/-1HB26ko2H8/hqdefault.jpg");
    private Optional<String> companyName = Optional.of("Amantido");
    private Optional<String> companyUrl = Optional.absent();
    private Optional<String> personalUrl = Optional.of("https://en.wikipedia.org/wiki/Banana_Joe_(film)");
    private Optional<String> twitterUsername = Optional.of("@bananaJoe1982");

    public static SpeakerFixtures aSpeaker() {
        return new SpeakerFixtures();
    }

    private SpeakerFixtures() {
        // Not instantiable
    }

    public SpeakerFixtures withId(String id) {
        this.id = id;
        return this;
    }

    public SpeakerFixtures withNumericId(long numericId) {
        this.numericId = numericId;
        return this;
    }

    public SpeakerFixtures withName(String name) {
        this.name = name;
        return this;
    }

    public SpeakerFixtures withBio(String bio) {
        this.bio = bio;
        return this;
    }

    public SpeakerFixtures withCompanyName(Optional<String> companyName) {
        this.companyName = companyName;
        return this;
    }

    public SpeakerFixtures withCompanyUrl(Optional<String> companyUrl) {
        this.companyUrl = companyUrl;
        return this;
    }

    public SpeakerFixtures withPersonalUrl(Optional<String> personalUrl) {
        this.personalUrl = personalUrl;
        return this;
    }

    public SpeakerFixtures withPhotoUrl(Optional<String> photoUrl) {
        this.photoUrl = photoUrl;
        return this;
    }

    public SpeakerFixtures withTwitterUsername(Optional<String> twitterUsername) {
        this.twitterUsername = twitterUsername;
        return this;
    }

    public Speaker build() {
        return Speaker.create(
                id,
                numericId,
                name,
                bio,
                companyName,
                companyUrl,
                personalUrl,
                photoUrl,
                twitterUsername
        );
    }
}

File: app/src/test/java/net/squanchy/speaker/domain/view/SpeakerFixtures.kt
package net.squanchy.speaker.domain.view

import arrow.core.Option

fun aSpeaker(
    id: String = "bananana",
    numericId: Long = 5466L,
    name: String = "Banana Joe",
    bio: String = """Joe ¡oh Banana Joe!
Tu tienes, ¡oh Banana nana Joe!
Corazón gigante y alma soñan,te
¡Oh Banana nana Joe!
Joe ¡oh Banana Joe!
Tu eres, ¡oh Banana na na Joe!
Un gran marinero con puños de acero""",
    photoUrl: Option<String> = Option("https://i.ytimg.com/vi/-1HB26ko2H8/hqdefault.jpg"),
    companyName: Option<String> = Option("Amantido"),
    companyUrl: Option<String> = Option("http://banana.joe"),
    personalUrl: Option<String> = Option("https://en.wikipedia.org/wiki/Banana_Joe_(film)"),
    twitterUsername: Option<String> = Option("@bananaJoe1982")
) = Speaker(
    id = id,
    numericId = numericId,
    name = name,
    bio = bio,
    photoUrl = photoUrl,
    companyName = companyName,
    companyUrl = companyUrl,
    personalUrl = personalUrl,
    twitterUsername = twitterUsername
)
